File: Domain/SpecialtyCore.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class SpecialtyCore
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public ICollection<ISCEDCore> ISCEDCores { get; set; } = new List<ISCEDCore>();
        public ICollection<Specialty> Specialties { get; set; } = new List<Specialty>();
    }
}

File: client-app/src/app/models/specialtyCore.ts
import { iscedCore } from "./iscedSpecialty";

export interface SpecialtyCore {
    id: string;
    name: string;
    iscedCores: iscedCore[];
}

export class SpecialtyCore implements SpecialtyCore {
    constructor(init?: SpecialtyCore | Partial<SpecialtyCore>) {
        Object.assign(this, init);
    }
}
